testnet/stacks-node/src/config.rs

```diff
@@ -1927,7 +1927,42 @@ impl NodeConfig {
         let pubkey = Secp256k1PublicKey::from_hex(pubkey_str)
             .unwrap_or_else(|_| panic!("Invalid public key '{pubkey_str}'"));
         debug!("Resolve '{}'", &hostport);
-        let sockaddr = hostport.to_socket_addrs().unwrap().next().unwrap();
+
+        let mut attempts = 0;
+        let max_attempts = 5;
+        let mut delay = Duration::from_secs(2);
```

Comment:
Nit: It feels a bit weird to me to hard-code these parameters here. I would be more happy to see a solution which breaks out the retry logic to a separate function, similar to `https://docs.rs/retry/2.0.0/retry/` while allowing the user to configure the retry.

Not sure if it's worth the effort though, but it would be nice to have this testable and tested in the code which the current change doesn't seem to be.

Reply:
sure, I can make it optionally configurable if it's okay for me to expand `NodeConfigFile`
As for testing it, I wouldn't be so sure since I tested it manually by just turning off my wifi before that method gets called.

Reply:
Yeah `NodeConfigFile` looks like a good home for a potential `network_retry_timeout` parameter (just a single parameter in my opinion from which the `attempts`, `max_attempts` and `delay` can be derived). Although that's debatable, so perhaps best would be to merge this as-is since it's already got the necessary approvals and do that work as a follow-up if there's time and interest.

Reply:
> As for testing it, I wouldn't be so sure since I tested it manually by just turning off my wifi before that method gets called.

Yeah that's what I suspected. I don't see any straightforward way to make the Ctrl-C shutdown logic testable given how the code is currently structured, so I guess that'll have to do for now.

Reply:
Cool, will ask @jferrant to merge after the rebase finishes.